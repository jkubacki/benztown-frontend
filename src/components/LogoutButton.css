@media all and (max-width:767px) {
  .btn-block-sm { width: 100%; display: block; }
}
