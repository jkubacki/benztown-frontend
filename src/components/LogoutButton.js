import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { logout as logoutAction } from 'actions/auth';

import './LogoutButton.css';

function LogoutButton({ logout }) {
  return (
    <button type="button" onClick={logout} className="btn btn btn-outline-dark btn-block-sm">
      Logout
    </button>
  );
}

LogoutButton.propTypes = {
  logout: PropTypes.func.isRequired,
};

export { LogoutButton as LogoutButtonUnwrapped };

export default compose(
  connect(
    null,
    { logout: logoutAction },
  ),
)(LogoutButton);
