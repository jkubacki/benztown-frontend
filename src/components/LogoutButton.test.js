import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { LogoutButtonUnwrapped as LogoutButton } from './LogoutButton';

const props = { logout: jest.fn(() => Promise.resolve()) };

describe('LogoutButton', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<LogoutButton {...props} />);
  });

  it('should render corectly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render action button', () => {
    expect(wrapper.find('button').length).toEqual(1);
  });

  it('should call logout when button is clicked', () => {
    const logoutButton = wrapper.find('button');
    logoutButton.simulate('click');

    expect(props.logout).toHaveBeenCalled();
  });
});
