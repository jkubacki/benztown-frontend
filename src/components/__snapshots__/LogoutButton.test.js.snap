// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutButton should render corectly 1`] = `
<button
  className="btn btn btn-outline-dark btn-block-sm"
  onClick={[MockFunction]}
  type="button"
>
  Logout
</button>
`;
